js/walutomat.js

```diff
@@ -0,0 +1,272 @@
+                'private': {
+                    'get': [
+                        'account/id',
+                        'account/balances',
+                        'account/history',
+                        'market/orders',
+                        'market/orders/{orderId}',
+                    ],
+                    'post': [
+                        'market/orders',
+                        'market/orders/close/{orderId}',
+                    ],
+                },
+            },
+            'exceptions': {
+                'INVALID ARGUMENT': ExchangeError,
+                'INVALID PARAMETER': ExchangeError,
+                'INSUFFICIENT_FUNDS': InsufficientFunds,
+                'MISSING_API_KEY': AuthenticationError,
+            },
+        });
+    }
+
+    fetchMarkets (params = {}) {
+        let currencies = ['EUR', 'GBP', 'USD', 'CHF', 'PLN'];
+        let markets = [];
+        let currenciesLength = currencies.length;
+        for (let base = 0; base < currenciesLength - 1; base++) {
+            let newIndex = base + 1;
+            for (let quote = newIndex; quote < currencies.length; quote++) {
+                markets.push ({
+                    'id': currencies[base] + '_' + currencies[quote],
+                    'symbol': currencies[base] + '/' + currencies[quote],
+                    'base': currencies[base],
+                    'quote': currencies[quote],
+                    'active': true,
+                    'precision': {
+                        'amount': 2,
+                        'price': 4,
+                    },
+                    'info': currencies[base] + ' - ' + currencies[quote] + ' exchange',
+                });
+            }
+        }
+        return markets;
+    }
+
+    async fetchBalance (params = {}) {
+        let balances = await this.privateGetAccountBalances ();
+        let result = {};
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            result[this.safeString (balance, 'currency')] = {
+                'free': this.safeFloat (balance, 'balanceAvailable'),
+                'used': this.safeFloat (balance, 'balanceReserved'),
+                'total': this.safeFloat (balance, 'balanceAll'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        let symbolObject = { 'symbol': this.formatSymbol (symbol) };
+        let response = await this.publicGetPublicMarketOrderbookSymbol (this.extend (symbolObject, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'baseVolume');
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        let request = {
+            'operationType': 'MARKET_FX',
+            'currencies': symbol,
+            'continueFrom': since,
+            'volume': limit,
+            'sortOrder': params.sortOrder || 'DESC',
```

Comment:
Ok, now it's a different issue in this line, same thing with assoc keys, basically...
Unfortunately it doesn't transpile that way... And in this particular place there's a reduncancy – if you look at line 122 – it extends the request with params['sortOrder'] (if any) there already, so, we don't even need to do that in the `request = ` assignment. So to work in other languages it simply to be like so:

```JavaScript
let request = {
    ...,
    'sortOrder': 'DESC', // default is DESC, can be overrided later in line 122 with "params"
};
```

Reply:
Yeah, in Python || will be changed to OR and will work, but PHP :) I will fix it right now :)